### PR TITLE
Plaquette double copy bug fixed + More minor updates

### LINKED ISSUE #13: Creating new plaquettes by copy-drag spawns extra plaquettes
When I copy a plaquette by dragging, sometimes an extra plaquette appears, usually next to the new one, but sometimes on a different part of the tileset. [Example video](https://clipchamp.com/watch/NNNNwXJl3lF)

Mentioned in #12.

### PR DESCRIPTION
**Pull request by smburdick**
Fixed #13 + minor formatting updates

### CODE REVIEW

**Comment by smburdick** [comment]
Fixed #13

File: tqecjs/src/workspace/plaquettes/PlaquetteClass.js [hunk 1]
```diff
@@ -329,18 +335,13 @@ export default class Plaquette extends Graphics {
...
-			if (currentTime - lastClickTime < 300) {
-				// Set the alpha of the plaquette to 0.8 to indicate
-				// Check if the plaquette is clicked, not the button
-				this.showButton = !this.showButton;
-				// this.showButton ? (this.alpha = 0.8) : (this.alpha = 1);
-				// Show or hide the buttons
-				this.controlPanel.visible = this.showButton;
+			if (currentTime - this.lastClickTime < 300) {
+				// Show or hide the control panel
+				this.controlPanel.visible = !this.controlPanel.visible;
```
Comment (rryoung98) [inline tqecjs/src/workspace/plaquettes/PlaquetteClass.js:342]: Thanks for simplifying

File: tqecjs/src/workspace/plaquettes/PlaquetteClass.js [hunk 2]
```diff
@@ -216,26 +215,32 @@ export default class Plaquette extends Graphics {
...
+		console.log(`Drag (dx, dy) = (${deltaX}, ${deltaY})`);
 		let shiftQ = null;
-		if (
-			this.isDragging &&
-			(deltaX >= this.gridSize * 0.4 || deltaY >= this.gridSize * 0.4)
-		) {
+		const scaleFactor = 0.4;
+		const threshold = this.gridSize * scaleFactor;
+		// If the cursor is moved meaningfully, check which direction to copy to
+		if (Math.abs(deltaX) >= threshold || Math.abs(deltaY) >= threshold) {
```
Comment (rryoung98) [inline tqecjs/src/workspace/plaquettes/PlaquetteClass.js:237]: Although there are times when the canvas continues to make plaquettes after the user has stopped cliccking down on the mouse pad and is just dragging their cursor, the new changes do make things smoother

File: tqecjs/src/workspace/plaquettes/PlaquetteClass.js [hunk 3]
```diff
@@ -297,25 +301,27 @@ export default class Plaquette extends Graphics {
...
+		}
+		console.log("New plaquette color = " + newColor.description);
+		const newPlaquette = new Plaquette(
+			newQubits,
+			this.workspace,
+			this.gridSize,
+			newColor
+		);
+		// Add the plaquette to parent container
+		this.parent.addChild(newPlaquette);
```
Comment (rryoung98) [inline tqecjs/src/workspace/plaquettes/PlaquetteClass.js:322]: Adding to the parent for the first plaquette would add it to the tileClass, but consequent plaquettes would still be in the child plaquette. This causes issues when clearing as mentioned in #29

File: tqecjs/src/workspace/plaquettes/PlaquetteClass.js [hunk 4]
```diff
@@ -368,8 +369,9 @@ export default class Plaquette extends Graphics {
 		this.controlPanel.addChild(this.colorButton);
 	};
 
-	clearPlaquette = () => {
+	initializeClearButton = () => {
 		this.clearButton.on('click', (_event) => {
+			console.log("Destroying plaquette")
```
Comment (rryoung98) [inline tqecjs/src/workspace/plaquettes/PlaquetteClass.js:374]: The destroying feature still needs a bit of work and I've spun it out into an issue: [#29](https://github.com/QCHackers/tqec/issues/24)

**Review by rryoung98 — COMMENTED** [review]
Looks good, sorry for the wait

**Review by rryoung98 — APPROVED** [review]

**Pull request merged**